base_classes/NXtransformations.nxdl.xml

```diff
@@ -108,6 +116,93 @@
+		* *transmission*:
+			* point detector in the beam
+			* 20 cm downstream from the sample
+		* *vertical*:
+			* point detector 10 cm downstream from the sample
+			* making an angle of 5 degrees with the beam w.r.t. the sample
+			* positioned in the XZ-plane above the XY-plane
+		* *horizontal*:
+			* point detector 11 cm downstream from the sample
+			* making an angle of 6 degrees with the beam w.r.t. the sample
+			* positioned in the XY-plane above the XZ-plane
+
+		.. code-block::
+
+		  entry:NXentry
+		    coordinate_system:NXtransformations
+		      beam=NaN                   # value is never used
+		        @depends_on=gravity
+		        @vector=[1 0 0]          # X-axis points in the beam direction
+		      gravity=NaN                # value is never used
+		        @depends_on=.
+		        @vector=[0 0 -1]         # Z-axis points up
+		    sample:NXsample
+		      depends_on=diffr/phi
+		      diffr:NXtransformations
+		        phi=45
+		          @depends_on=chi
+		          @transformation_type=rotation
+		          @vector=[0 1 0]
+		          @units=degrees
+		        chi=90
+		          @depends_on=rotation_angle
+		          @transformation_type=rotation
+		          @vector=[0 0 1]
+		          @units=degrees
+		        rotation_angle=30
+		          @depends_on=../../coordinate_system/beam
+		          @transformation_type=rotation
+		          @vector=[0 1 0]
+		          @units=degrees
+		    instrument:NXinstrument
+		      vertical:NXdetector
+		        depends_on=position/azimuth
+		        position:NXtransformations
+		          azimuth=0              # stay in the vertical plane
+		            @depends_on=polar
+		            @units=degrees
+		            @vector=[-1 0 0]
+		          polar=5                # title above the horizontal plane
+		            @depends_on=distance
+		            @units=degrees
+		            @vector=[0 -1 0]
+		          distance=10            # move downstream from the sample
+		            @depends_on=../../coordinate_system/beam
+		            @units=cm
+		            @vector=[1 0 0]
+		      horizontal:NXdetector
+		        depends_on=position/azimuth
+		        position:NXtransformations
+		          azimuth=90             # rotate to the horizontal plane
+		            @depends_on=polar
+		            @units=degrees
+		            @vector=[-1 0 0]
+		          polar=6                # title above the horizontal plane
+		            @depends_on=distance
+		            @units=degrees
+		            @vector=[0 -1 0]
+		          distance=11            # move downstream from the sample
+		            @depends_on=../../coordinate_system/beam
+		            @units=cm
+		            @vector=[1 0 0]
+		      transmission:NXdetector
+		        depends_on=position/azimuth
+		        position:NXtransformations
+		          distance=20            # move downstream from the sample
+		            @depends_on=../../coordinate_system/beam
+		            @units=cm
+		            @vector=[1 0 0]
+
```

Comment:
Add explicit matrices for calculating detector positions:
```
r_lab = T_x(dist) R_y(-pol) R_x(-azi) r_det
```
where T<sub>x</sub> is translation in x, R<sub>x</sub> is rotation about x, etc.

In fact, this shows that the `depends_on` chain is wrong and should be `detector` depends on `distance` (as the detector is mounted on an arm of that length) depends on `polar` depends on `azimuth` depends on `.` so that
```
r_lab =  R_x(-azi) R_y(-pol) T_x(dist) r_det
```